protocols/virtualfund/virtualfund.go

```diff
@@ -57,6 +59,66 @@ func (c *Connection) Equal(d *Connection) bool {
 
 }
 
+// ledgerChannelAffordsExpectedGuarantees returns true if, for the channel inside the connection, and for each asset keying the input variables, the channel can afford the allocation given the funding.
+// The decision is made based on the latest supported state of the channel.
+//
+// Both arguments are maps keyed by the same asset.
+func (c *Connection) ledgerChannelAffordsExpectedGuarantees() bool {
+	return c.Channel.Affords(c.getExpectedGuarantees(), c.Channel.OnChainFunding) // todo: #420 deprecate
+}
+
+// insertGuaranteeInfo mutates the reciever Connection struct.
+func (c *Connection) insertGuaranteeInfo(a0 types.Funds, b0 types.Funds, vId types.Destination, left types.Destination, right types.Destination) error {
+
+	c.GuaranteeInfo = GuaranteeInfo{
+		Left:                 left,
+		Right:                right,
+		LeftAmount:           a0,
+		RightAmount:          b0,
+		GuaranteeDestination: vId,
+	}
+
+	// Check that the guarantee metadata can be encoded. This allows us to avoid clunky error-return-chains for getExpectedGuarantees
+	metadata := outcome.GuaranteeMetadata{
+		Left:  c.GuaranteeInfo.Left,
+		Right: c.GuaranteeInfo.Right,
+	}
+	_, err := metadata.Encode()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getExpectedGuarantees returns a map of asset addresses to guarantees for a Connection.
+func (c *Connection) getExpectedGuarantees() map[types.Address]outcome.Allocation {
+	expectedGuaranteesForLedgerChannel := make(map[types.Address]outcome.Allocation)
+	metadata := outcome.GuaranteeMetadata{
+		Left:  c.GuaranteeInfo.Left,
+		Right: c.GuaranteeInfo.Right,
+	}
+	encodedGuarantee, err := metadata.Encode()
+	// This error is unexpected. insertGuaranteeInfo checks that the guarantee metadata can be encoded.
+	// If this panic is triggered, GuranteeInfo has been modified after creation
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Minor comment: Currently, the check on L87 can be removed, breaking the assumption you're making here. A less brittle way of doing what you want is to store the encoded guarantee on L86 above, and checking explicitly whether the encoded metadata on L101 matches what's stored.

(This code is temporary, since `virtualFund` won't be responsible for inspecting `outcome.Exit` items in the future. So, we don't have to worry about this issue.)

Reply:
ah, I see this is copy/pasted anyway!